Added eventing tutorials to docs (#14706)
kyma-project/kyma: docs/03-tutorials/00-eventing/evnt-02-subs-with-multiple-filters.md

---
title: Create Subscription subscribing to multiple event types
---

The [Subscription](../../05-technical-reference/00-custom-resources/evnt-01-subscription.md) custom resource definition (CRD) is used to subscribe to events. In this tutorial, you learn how to subscribe to one or more event types using the Kyma Subscription.

## Prerequisites

> 1. Follow the [prerequisites steps](../../02-get-started/04-trigger-workload-with-event.md#prerequisites) mentioned in the getting-started guide.

Reviewer: ```suggestion
1. Follow the [prerequisites steps](../../02-get-started/04-trigger-workload-with-event.md#prerequisites) in the Getting Started guide.
```

---

> title: Create subscription subscribing to multiple event types

Author: Updated it to `Subscription`

---

The [Subscription](../../05-technical-reference/00-custom-resources/evnt-01-subscription.md) custom resource definition (CRD) is used to subscribe to events. In this tutorial, we will show how to subscribe to one or more event types using the Kyma Subscription.

> ## Prerequisites

Reviewer: The prerequisites section is identical for all four tutorials. Will you remember updating them all if any prerequisite changes? You might want to extract the prerequisites to maintain them just once and link to that single source of truth. Basically, a three-step instruction "Set up Eventing" or something like that. Or keep them just in the first tutorial and link to that in the others, something like that:

> ## Prerequisites:
> You have performed the prerequisite steps in tutorial [blabla](xyz).

1. Provision a [Kyma Cluster](../../02-get-started/01-quick-install.md).
2. (Optional) Deploy [Kyma Dashboard](../../01-overview/main-areas/ui/ui-01-gui.md) on the Kyma cluster using the following command. Alternatively, you can also use `kubectl` CLI.
   ```bash
   kyma dashboard
   ```
3. (Optional) Install [CloudEvents Conformance Tool](https://github.com/cloudevents/conformance) for publishing events. Alternatively, you can also use `curl` to publish events.
   ```bash
   go install github.com/cloudevents/conformance/cmd/cloudevents@latest
   ```

## Create a Workload

First, create a sample Function that prints out the received event to console:

<div tabs name="Deploy a Function" group="create-workload">
  <details open>
  <summary label="Kyma Dashboard">
  Kyma Dashboard
  </summary>

1. Go to **Namespaces** and select the default Namespace.
2. Go to **Workloads** > **Functions** and click **Create Function +**.
3. Name the Function `lastorder` and click **Create**.

> 4. In the inline editor for the Function, modify its source replacing it with this code:

Reviewer: ```suggestion
4. In the inline editor for the Function replace its source with the following code:
```
For consistency with the other tutorials

In this example, we'll port-forward the Kyma Eventing Service to localhost.

1. Port-forward the Kyma Eventing Service to localhost. We will use port `3000`. Run:
   ```bash
   kubectl -n kyma-system port-forward service/eventing-event-publisher-proxy 3000:80
   ```
2. Publish an event of type `order.received.v1` to trigger your function. In another Terminal window run:

    <div tabs name="Publish an event" group="trigger-workload">
      <details open>
      <summary label="CloudEvents Conformance Tool">
      CloudEvents Conformance Tool
      </summary>
    
       ```bash
       cloudevents send http://localhost:3000/publish \
          --type sap.kyma.custom.myapp.order.received.v1 \
          --id cc99dcdd-6f6d-43d6-afef-d024eb276584 \
          --source myapp \
          --datacontenttype application/json \
          --data "{\"orderCode\":\"3211213\", \"orderStatus\":\"received\"}" \
          --yaml
       ```
    
      </details>
      <details>
      <summary label="curl">
      curl
      </summary>
    
       ```bash
       curl -v -X POST \
            -H "ce-specversion: 1.0" \
            -H "ce-type: sap.kyma.custom.myapp.order.received.v1" \

>             -H "ce-source: /default/io.kyma-project/custom" \

Reviewer 2: why is the source here a different source than in the conformance tool command?

Author: updated

---
title: Create Subscription subscribing to multiple event types
---

The [Subscription](../../05-technical-reference/00-custom-resources/evnt-01-subscription.md) custom resource definition (CRD) is used to subscribe to events. In this tutorial, you learn how to subscribe to one or more event types using the Kyma Subscription.

## Prerequisites

1. Follow the [prerequisites steps](../../02-get-started/04-trigger-workload-with-event.md#prerequisites) mentioned in the getting-started guide.

> 2. Create a Function by following the [instructions](../../02-get-started/04-trigger-workload-with-event.md#create-a-function) given in the getting-started guide.

Reviewer: ```suggestion
2. Create a Function by following the [instructions](../../02-get-started/04-trigger-workload-with-event.md#create-a-function) in the Getting Started guide.
```

## Verify the event delivery

> To verify that the events were properly delivered, check the logs of the Function by following the [instructions](../../02-get-started/04-trigger-workload-with-event.md#verify-the-event-delivery) given in the getting-started guide.

Reviewer: ```suggestion
To verify that the events were properly delivered, check the logs of the Function (see [Verify the event delivery](../../02-get-started/04-trigger-workload-with-event.md#verify-the-event-delivery)).
```
